### ISSUE TITLE
ENH: convert column of WKT strings to GeoSeries

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/geopandas#603]
xref https://github.com/geopandas/geopandas/issues/588#issuecomment-337720124 and https://github.com/geopandas/geopandas/issues/602#issue-269282586

If you have geometry strings, you can convert it to actualy geometries with shapely:

```
In [59]: import shapely.wkt

In [62]: s = pd.Series(["POINT (-38.5554 -3.73784)", "POINT (-38.5554 -3.73784)"])

In [64]: s.apply(shapely.wkt.loads)
```

We should at least add this to the docs, but would also be useful to have a helper function to do this in geopandas (and in the cython branch, this could also be done vectorized). We just need to think about a good API.

**Comment by jdmcbr** [comment]
How about a `from_str` method on geoseries? Analogous to the `from_` methods on different dtypes on dataframes or geodataframes (as well as on series, though I'd never actually used those myself). And if the `geometry` column for a user initializing a `GeoDataFrame` is a str, apply `shapely.wkt.loads` by default?

**Comment by jorisvandenbossche** [comment]
or `from_wkt` (since it has to be a specific string format, although `from_str` might be more discoverable with people not familiar with the WKT terminology)

Relatedly, note that I added a `from_wkt` function in the cython branch: https://github.com/geopandas/geopandas/pull/655. This does not yet provide a nice user interface, although in principle you could do: something like `GeoDataFrame(df, geometry=from_wkt(strings))`

**Comment by jdmcbr** [comment]
Oops, I'd forgotten about `from_wkt` on the cython branch. I leaned towards `from_str` for discoverability, as you noted, and inability to think of an alternative case in which a geometry would be stored as a string, though maybe that's just lack of imagination on my part.

**Comment by martinfleis** [comment]
Closed by #1710
